ssh.go

```diff
@@ -108,21 +108,33 @@ func initAuthMethod() {
 	authMethod = ssh.PublicKeys(signers...)
 }
 
+// SSHDialFunc can dial an ssh server and return a client
+type SSHDialFunc func(net, addr string, config *ssh.ClientConfig) (*ssh.Client, error)
+
 // Connect creates SSH connection to a specified host.
 // It expects the host of the form "[ssh://]host[:port]".
-// TODO: Split Signers to its own method.
 func (c *SSHClient) Connect(host string) error {
+	return c.ConnectWith(host, ssh.Dial)
+}
+
+// ConnectWith creates a SSH connection to a specified host. It will use dialer to establish the
+// connection.
+// TODO: Split Signers to its own method.
+func (c *SSHClient) ConnectWith(host string, dialer SSHDialFunc) error {
+
+	fmt.Println("connecting", host)
```

Comment:
Thanks for this PR- awesome stuff!

can you remove this debug Println. Or we should have a sup verbose mode, and have a logger there, etc. For now, remove this line.